### PR TITLE
Update storage plus docs / Remove `MultiIndex` PK default

### LINKED ISSUE #669: Fix `MultiIndex` last type param default / docs
With the latest implementation of type-safe bounds, the definition of a `MultiIndex` `PK` type is crucial, as it now serves to define the type of the (partial) bounds over the index key (the part that corresponds to the pk, that is). See https://github.com/confio/poe-contracts/pull/114/files#r812029191 for reference / an example.

Let's fix the docs to reflect this use case, and its importance.

Also, let's remove the default over `PK` that we introduced for backwards compatibility, as this type definition is now really important for specifying (type-safe) bounds.

### PR DESCRIPTION
**Pull request by maurolacy**
Closes #669.

### CODE REVIEW

File: packages/storage-plus/README.md [hunk 1]
```diff
@@ -211,22 +211,25 @@ legitimate address, and not random text which will fail later.
...
 
-Here how we use it with composite keys. Just define a tuple as a key and use that
+Here's how we use it with composite keys. Just define a tuple as a key and use that
 everywhere you used a byte slice above.
 
 ```rust
-// Note the tuple for primary key. We support one slice, or a 2 or 3-tuple
-// adding longer tuples is quite easy but unlikely to be needed.
+// Note the tuple for primary key. We support one slice, or a 2 or 3-tuple.
+// Adding longer tuples is possible, but unlikely to be needed.
```
Comment (ueco-jb) [inline packages/storage-plus/README.md:232]: :) Points for honesty

File: packages/storage-plus/README.md [hunk 2]
```diff
@@ -314,7 +317,7 @@ fn demo() -> StdResult<()> {
 
 In addition to getting one particular item out of a map, we can iterate over the map
 (or a subset of the map). This let us answer questions like "show me all tokens",
-and we provide some nice [`Bound`](#Bound) helpers to easily allow pagination or custom ranges.
+and we provide some nice [`Bound`](#bound) helpers to easily allow pagination, or custom ranges.
```
Comment (ueco-jb) [inline packages/storage-plus/README.md:320]: Actually I think it should be without ','.

File: packages/storage-plus/README.md [hunk 3]
```diff
@@ -323,7 +326,7 @@ over all items with `range(store, min, max, order)`. It supports `Order::Ascendi
 `min` is the lower bound and `max` is the higher bound.
 
 If the `min` and `max` bounds are `None`, `range` will return all items under the prefix. You can use `.take(n)` to
-limit the results to `n` items and start doing pagination. You can also set the `min` bound to
+limit the results to `n` items, and start doing pagination. You can also set the `min` bound to
```
Comment (ueco-jb) [inline packages/storage-plus/README.md:329]: As above.

**Review by ueco-jb — APPROVED** [review]
lgtm
I salute you for your commitment, since this PR could be a one liner after all.

File: packages/storage-plus/README.md (hunk 1 above)
Comment (maurolacy) [reply packages/storage-plus/README.md:232]: :)

**Pull request merged**